src/AllCocktailsPage/AllCocktailsPage.tsx

```diff
@@ -1,21 +1,22 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useRef } from "react";
 import { AllCocktailsDetails } from '../App/App';
-import { getAllCocktails } from "../apiCalls";
 import CocktailCard from '../CocktailCard/CocktailCard';
 import './AllCocktailsPage.scss';
 
 export interface AllCocktailsProps {
-  allCocktails: AllCocktailsDetails[];
+	givenCocktails: AllCocktailsDetails[];
 }
 
 const AllCocktailsPage: React.SFC<AllCocktailsProps> = (props) => {
-  const [allCocktails, setAllCocktails] = useState(props.allCocktails);
-
+  const [givenCocktails, setGivenCocktails] = useState(props.givenCocktails);
   const [error, setError] = useState<string>("");
 
-  const cocktailsArray = Object.values(allCocktails);
-  
-  const cocktailCards = cocktailsArray.map(cocktail => {
+	useEffect(() => {
+		setGivenCocktails(props.givenCocktails)
+	}, [props]);
```

Comment:
This block acts as componentDidUpdate essentially to allow AllCocktails to unmount and remount between displaying All, or Favorites/Logged (short article: https://blog.bitsrc.io/replacing-react-lifecycle-methods-with-hooks-a-comprehensive-guide-dfd5cbe1f274)

Reply:
interesting to see it used this way! thanks for linking the article

Reply:
This is what we were talking about yesterday.  Perfect Becca.